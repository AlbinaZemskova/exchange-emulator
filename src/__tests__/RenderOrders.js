import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';
import {
  RenderOrders,
  mapStateToProps,
} from '../RenderOrders';
import 'jest-prop-type-error';

Enzyme.configure({ adapter: new Adapter() });

describe('<RenderOrder />', () => {
  let wrapper;
  let buyOrders;
  let sellOrders;

  beforeEach(() => {
    buyOrders = [
      {
        price: 1000,
        count: 100,
      },
      {
        price: 1000,
        count: 200,
      },
      {
        price: 2000,
        count: 1,
      },
    ];

    sellOrders = [
      {
        price: 5000,
        count: 300,
      },
      {
        price: 5000,
        count: 250,
      },
      {
        price: 17000,
        count: 300,
      },
    ];

    wrapper = shallow(
      <RenderOrders
        buyOrders={buyOrders}
        sellOrders={sellOrders}
      />,
    );
  });

  describe('RenderOrders component', () => {
    describe('render', () => {
      it('renders without crashing', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
      });

      it('checking correct proptypes', () => {
        expect(() => (
          <RenderOrders
            buyOrders={[
              {
                price: '1',
                count: '2',
              },
            ]}
            sellOrders={[
              {
                price: '1',
                count: '2',
              },
            ]}
          />
        )).toThrow();
      });
    });
  });

  describe('mapStateToProps', () => {
    it('should return an object with the arrays of objects', () => {
      const mockState = {
        buyOrders: {
          buyOrders,
        },
        sellOrders: {
          sellOrders,
        },
      };

      const expected = {
        buyOrders,
        sellOrders,
      };

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });
});
