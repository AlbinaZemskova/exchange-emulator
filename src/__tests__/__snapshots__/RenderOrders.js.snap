// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RenderOrder /> RenderOrders component render renders without crashing 1`] = `
<Fragment>
  <Container
    className="d-flex justify-content-center"
    tag="div"
  >
    <Row
      tag="div"
    >
      <Col
        lg="12"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <ListOfOrders
          listOfOrders={
            Array [
              Object {
                "count": 300,
                "price": 17000,
              },
              Object {
                "count": 250,
                "price": 5000,
              },
              Object {
                "count": 300,
                "price": 5000,
              },
            ]
          }
        />
        <hr
          className="font-weigth-bold bg-dark mr-2 w-60"
        />
        <ListOfOrders
          listOfOrders={
            Array [
              Object {
                "count": 100,
                "price": 1000,
              },
              Object {
                "count": 200,
                "price": 1000,
              },
              Object {
                "count": 1,
                "price": 2000,
              },
            ]
          }
        />
      </Col>
    </Row>
  </Container>
  <hr />
</Fragment>
`;
